addons/mail/static/src/attachments/attachment_model.js

```diff
@@ -90,8 +90,9 @@ export class Attachment {
 
     get defaultSource() {
         const route = url(this.urlRoute, this.urlQueryParams);
+        const encodedRoute = encodeURIComponent(route + '#pagemode=none')
```

Comment:
`encodeURIComponent("#")` escapes it, which will mistakenly remove this option.
So you should not `encodeURIComponent` `#` options, only the route should be encoded